MySql-JSONify 2.0

Reqrefusion opened the pull request:
Add:
*MultiSelect
-Use comma(col1,col2)
*MultiOrder
-Use comma(col1,asc or asc,col1)
-Use semicolon(col1,asc;desc;col3)
*Filter
*MultiFilter
-Use comma(searchValue or searchValue,col2,Like or  searchValue,col2,not like)
-Use semicolon(%searchValue%,not like;searchValue2;like or searchValue;searchValue1 or %searchValue%,not like,col2,and;searchValue2,like,col3 )
*Page System
*İnfo Array
*Vanilla Jquery Table*
*Old Php Version Support*

*On Progress

Reqrefusion committed:
MySql-JSONify 2.0

Add:
*MultiSelect
*MultiOrder
*Filter
*MultiFilter
*Page System
*İnfo Array
*Vanilla Jquery Table*
*Old Php Version Support*

Nicklas766 commented:
Hi @Reqrefusion!

Thank you for your contribution. Be aware that this is a very old project and should be rewritten.

I'm impressed that you seem to have managed to add new functionality (since the project is from the start poorly written). By looking at your pull request, I think the next step would be to:

- Decrease the complexity in the `filterOrganizer` and `orderOrganizer` functions.

- Remove all commented out code.

- Trying to achieve a similar structure to something like [this](https://www.tutorialspoint.com/php/php_coding_standard.htm), but it might've looked different for you locally.

It seems like you have put effort into this, so I respectfully shared some ideas for improvements above. However, because this project is so old at the moment, I must decline this merge request. But please feel free to copy all the code and use it in whatever project you want.

Best regards,
Nicklas

Reqrefusion commented:
Hi @Nicklas766!
First of all, when I examined similar projects, this project was distinguished with simplicity. I personally fell in love with the simplicity of the project. I also have to rewrite the project. I have some questions in my mind when rewriting. The first question that comes to my mind is, should more than one class structure be preserved? Although I found it confusing at first, I liked this structure and I am for its protection.

- I realized that the functions I added(`filterOrganizer `and `orderOrganizer`) were complicated, I tried to simplify, so I added a few extra functions. In general, I will try to simplify. They might have been simpler, but I wrote extra codes to provide an easy spelling. Also, the slices and pieces I used came to my mind at the moment, but I can't say anything for them.

- I will remove the codes in the comments. However, I am not very good at adding English comments.

- I was careful about the code structure in some places. However, I cannot say that it fits the link you provided. I will be more careful about the code structure when rewriting.

I understand that you reject my merge request, but I really want to make this project more comprehensive. I have a personal love for projects written in Vanilla Php and this project can provide great convenience in ajax queries to be used for the website. I already added the table page to the project to show this.

Finally, you always mentioned the fixes, but what kind of a feature can be added to this project in your opinion. I added all the features I can think of.

I hope you continue to make a comeback this way.
Best regards,
Reqrefusion
